packages/react-composites/src/composites/CallComposite/components/CapabilitiesNotficationBar.tsx

```diff
@@ -0,0 +1,275 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/* @conditional-compile-remove(capabilities) */
+import {
+  CapabilitiesChangeInfo,
+  CapabilitiesChangedReason,
+  ParticipantCapabilityName,
+  ParticipantRole
+} from '@azure/communication-calling';
+/* @conditional-compile-remove(capabilities) */
+import { IMessageBarProps, MessageBar, MessageBarType, Stack } from '@fluentui/react';
+/* @conditional-compile-remove(capabilities) */
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+/* @conditional-compile-remove(capabilities) */
+import { useLocale } from '../../localization';
+/* @conditional-compile-remove(capabilities) */
+import { CallCompositeStrings } from '../Strings';
+
+/* @conditional-compile-remove(capabilities) */
+/**
+ * @beta
+ */
+export interface CapabilitiesChangeNotificationBarProps extends IMessageBarProps {
+  capabilitiesChangeInfo?: CapabilitiesChangeInfo;
+  participantRole?: ParticipantRole;
+}
+
+/* @conditional-compile-remove(capabilities) */
+/**
+ * @beta
+ */
+export interface CapabalityChangeNotification {
```

Comment:
There is a lot of logic here for a UI component.

Can we separate the data layer from the UI a bit more?
Have something in a lower level that collect the various capability issues and then maintains some data structure that the UI can read through a selector? When we dismiss from the UI we push the signal down to remove that notification from this data structure so if you don't dismiss the notification between screens it will still be showing the most important thing?

Reply:
Moved the logic out of the UI component and moved the management of the notification collection and dismissals in CallArrangement.